test/index.test.js

```diff
@@ -90,14 +90,57 @@ describe('Out of Band cache', () => {
 
     it('refreshes on an expired item', async function () {
       const fridge = new Cache({ maxAge: -10, maxStaleness: -10 }); // items are immediately stale
-      const getter = async i => i;
 
-      await fridge.get('old milk', {}, getter);
+      await fridge.get('old milk', {}, simpleGet);
       await sleep(100);
-      const expired = await fridge.get('old milk', { maxAge: 10, maxStaleness: 10 }, getter);
+      const expired = await fridge.get('old milk', { maxAge: 10, maxStaleness: 10 }, simpleGet);
 
       assume(expired.fromCache).is.falsey();
     });
+
+    it('does not add an item to the cache if preconfigured not to', async function () {
+      const dopey = new Cache({ shouldCache: () => false });
+
+      await dopey.get('diamond', {}, simpleGet);
+      assume(dopey._caches[0]._items).has.length(0);
+    });
+
+    it('does not add an item if we supply shouldCache as a get option', async function () {
+      const dopey = new Cache({});
+
+      await dopey.get('diamond', { shouldCache: () => false }, simpleGet);
+      assume(dopey._caches[0]._items).has.length(0);
+    });
+
+    it('does not add an item if it fails shouldCache', async function () {
+      const dopey = new Cache({ shouldCache: v => v !== 'words' });
+      // dopey doesn't know how to speak so he doesn't remember any words
+
+      await dopey.get('sounds', {}, simpleGet);
+      await dopey.get('words', {}, simpleGet);
+      await dopey.get('diamonds', {}, simpleGet);
+      assume(dopey._caches[0]._items).has.length(2);
+    });
+
+    it('only sets the cache values with shouldCache is a function', async function () {
+      const wonderland = new Cache({});
+
+      // default
+      await wonderland.get('tweedle-dee', {}, simpleGet);
+      assume(wonderland._caches[0]._items).has.length(1);
+
+      // boolean
+      await wonderland.get('tweedle-dum', { shouldCache: true }, simpleGet);
```

Comment:
These seem like dev-time errors that should result in exceptions throwing and not what is happening here and "oops we're not caching but didn't tell you about it at all"